arf/load_str.c

```diff
@@ -0,0 +1,149 @@
+/*
+    Copyright (C) 2019 Julian Rüth
+
+    This file is part of Arb.
+
+    Arb is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <http://www.gnu.org/licenses/>.
+*/
+
+#include <string.h>
+
+#include "arf.h"
+
+static const char* arf_serialized_zero = "0 0";
+static const char* arf_serialized_pos_inf = "0 -1";
+static const char* arf_serialized_neg_inf = "0 -2";
+static const char* arf_serialized_nan = "0 -3";
+
+char *
+arf_dump_str(const arf_t x)
+{
+    size_t res_len;
+    char * res;
+
+    if (arf_is_special(x))
+    {
+        const char* ret;
+        if (arf_is_zero(x))
+        {
+            ret = arf_serialized_zero;
```

Comment:
Here and below, we should malloc new space to return, because the user might mutate the returned string. Also, this looks like it's going to crash if you free() the returned string (why doesn't this cause test failures?).

Reply:
I don't understand. I `flint_malloc` after the all the `if`s and copy the content. I do not return the actual `ret`.

Reply:
Ah, good, I just missed that.